src/js/components/stock-movement-wizard/request/EditPage.jsx

```diff
@@ -1512,14 +1513,23 @@ class EditItemsPage extends Component {
    */
   isUserAllowedToAddItems() {
     const { requestedBy, origin, destination } = this.state.values;
-    const { currentUserId, currentLocationId } = this.props;
-    if (currentLocationId === origin?.id) {
+    const { currentUser, currentLocation } = this.props;
+    if (currentLocation?.id === origin?.id) {
       return true;
     }
-    return currentLocationId === destination?.id && requestedBy?.id === currentUserId;
+    return currentLocation?.id === destination?.id && requestedBy?.id === currentUser?.id;
```

Comment:
I think we should either move this part to the util you've created or move the content of your util here.
Your util also checks for the user's permissions and so does this util, so someone might be confused about the differences of those. Can we somehow make it more clear?
For example both here and in your util you check for the `requestedBy === currentUser` - this is another thing that could be maybe moved to a separate method or refactor it in a way, where this is checked only once.
```javascript
return currentLocation?.id === destination?.id && requestedBy?.id === currentUser?.id;
```

Reply:
I don't think it's even necessary to check the electronicType here, because this component is used clearly for the request workflow, so it must be electronicType - you won't edit the outbounds using this component. 
If you can, please spend a bit time on figuring out whether it is possible to somehow merge those methods (that are currently determining whether the "Edit request items" button should be visible or not).

It's nitpicky, but for me it feels weird that we have methods: `isAddingItemsAllowed` and `canEditRequest` separately. If we ever have to change anything in the future, we might be confused what the differences between them are.
If you don't figure out anything, we can try together.